lib/options.go

```diff
@@ -96,9 +96,24 @@ func (v *TLSVersions) isTLS13() bool {
 
 // A list of TLS cipher suites.
 // Marshals and unmarshals from a list of names, eg. "TLS_ECDHE_RSA_WITH_RC4_128_SHA".
-// BUG: This currently doesn't marshal back to JSON properly!!
 type TLSCipherSuites []uint16
 
+// MarshalJSON will return the JSON representation according to supported TLS cipher suites
+func (s *TLSCipherSuites) MarshalJSON() ([]byte, error) {
+	var suiteNames []string
+	for _, id := range *s {
+		if suiteName, ok := SupportedTLSCipherSuitesToString[id]; ok {
+			suiteNames = append(suiteNames, suiteName)
+		} else {
+			return nil, errors.Errorf("Unknown cipher suite id: %d", id)
+		}
+	}
+
+	marshaled, _ := json.Marshal(suiteNames)
+
+	return marshaled, nil
```

Comment:
Why not just `return json.Marshal(suiteNames)`?

Reply:
Good question, no reason not to! Will update this now